pkgs/desktops/gnome-3/3.20/core/gnome-shell/default.nix

```diff
@@ -15,7 +16,8 @@ stdenv.mkDerivation rec {
 
   buildInputs = with gnome3;
     [ gsettings_desktop_schemas gnome_keyring gnome-menus glib gcr json_glib accountsservice
-      libcroco intltool libsecret pkgconfig python3 libsoup polkit libcanberra gdk_pixbuf librsvg
+      libcroco intltool libsecret pkgconfig python3 python3Packages.pygobject3
+      libsoup polkit libcanberra gdk_pixbuf librsvg
       clutter networkmanager libstartup_notification telepathy_glib docbook_xsl docbook_xsl_ns
```

Comment:
The derivation  `gnome-shell` will have a Python packages tree. The dependency, will have its own Python packages tree. Python packages from one env/tree cannot be imported in the other. I think that is what we want.

The idea behind this is that we can get rid of the wrappers that set `PYTHONPATH`.